contract/models/account_analytic_account.py

```diff
@@ -206,10 +206,13 @@ def _create_invoice(self):
     @api.multi
     def recurring_create_invoice(self):
         """
-        Create invoices from contracts
+        Create invoices or sales from contracts
         :return: invoices created
         """
-        invoices = self.env['account.invoice']
+        # NOTE: Don't use a browse because here there are 2 types of records:
+        #       sale or invoice. A manual 'set' fix error of
+        #       Mixing apples and oranges: account.invoice().union(sale.order)
+        invoices = set()
```

Comment:
This is very ugly because is caused by an inherited module. @angelmoya you have provoked here a real inconsistency with your module contract_sale_generation. You can't change the API for returning either sale orders or invoices. You MUST change your module for not returning the other things. There are other modules that depend on the API that says that it returns a recordset of invoices, not other thing. Please fix it, Ángel.

Reply:
OK, I will check

Reply:
I'm checking.... I tried to use contract code, and not rewrite all functionality... 
But finally if sale_generation is not part of base module and other modules adapt to it, then would be better to create as independent functionality isolating invoicing and sales generation.

Reply:
#118